parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -163,14 +167,39 @@ downloadAzureCNI() {
     CNI_TGZ_TMP=${VNET_CNI_PLUGINS_URL##*/} # Use bash builtin ## to remove all chars ("*") up to the final "/"
     retrycmd_get_tarball 120 5 "$CNI_DOWNLOADS_DIR/${CNI_TGZ_TMP}" ${VNET_CNI_PLUGINS_URL} || exit $ERR_CNI_DOWNLOAD_TIMEOUT
 }
-
+{{if NeedsContainerd}}
 downloadContainerd() {
-    CONTAINERD_DOWNLOAD_URL="${CONTAINERD_DOWNLOAD_URL_BASE}cri-containerd-${CONTAINERD_VERSION}.linux-amd64.tar.gz"
+    # currently upstream maintains the package on a storage endpoint rather than an actual apt repo
+    CONTAINERD_DOWNLOAD_URL="https://mobyartifacts.azureedge.net/moby/moby-containerd/${CONTAINERD_VERSION}+azure/bionic/linux_amd64/moby-containerd_${CONTAINERD_VERSION}+azure-1_amd64.deb"
```

Comment:
can we ask caller to pass in CONTAINERD_VERSION as an argument? Global var is not a good practice IMO.

Reply:
will do. i notice that we do use a mix of globally scoped vars and positional args in methods. might be a good idea to clean those up in the future for clairty.

Reply:
yes, the global vars are some inheritance..

Reply:
added positional arg